x/mint/module/abci.go

```diff
@@ -2,80 +2,180 @@ package mint
 
 import (
 	"context"
+	"fmt"
 
-	cosmosMath "cosmossdk.io/math"
+	"cosmossdk.io/errors"
+	"cosmossdk.io/math"
 	"github.com/allora-network/allora-chain/x/mint/keeper"
+	"github.com/allora-network/allora-chain/x/mint/types"
 	sdk "github.com/cosmos/cosmos-sdk/types"
 )
 
+func UpdateEmissionRate(
+	ctx sdk.Context,
+	k keeper.Keeper,
+	params types.Params,
+	ecosystemBalance math.Int,
+) (
+	emissionPerTimestep math.Int,
+	emissionPerUnitStakedToken math.LegacyDec,
+	err error,
+) {
+	fmt.Println("Updating emission rate")
+	// Get the expected amount of emissions this block
+	networkStaked, err := keeper.GetNumStakedTokens(ctx, k)
+	fmt.Println("Network staked", networkStaked)
+	if err != nil {
+		return math.Int{}, math.LegacyDec{}, err
+	}
+	totalSupply := k.GetSupply(ctx).Amount
+	lockedSupply := keeper.GetLockedTokenSupply()
+	circulatingSupply := totalSupply.Sub(lockedSupply)
+	if circulatingSupply.IsNegative() {
+		return math.Int{}, math.LegacyDec{}, errors.Wrapf(
+			types.ErrNegativeCirculatingSupply,
+			"total supply %s, locked supply %s",
+			totalSupply.String(),
+			lockedSupply.String(),
+		)
+	}
+	fmt.Println("Total supply", totalSupply)
+	fmt.Println("Locked supply", lockedSupply)
+	fmt.Println("Circulating supply", circulatingSupply)
+	fmt.Println("FEmissionNumerator", params.FEmissionNumerator)
+	fmt.Println("FEmissionDenominator", params.FEmissionDenominator)
+	targetRewardEmissionPerUnitStakedToken,
+		err := keeper.GetTargetRewardEmissionPerUnitStakedToken(
```

Comment:
indentation is off

Reply:
it should be auto indented by the go formatter